src/ArithmeticProcedures.cpp

```diff
@@ -429,9 +439,18 @@ Object scheme::finitePEx(VM* theVM, int argc, const Object* argv)
 {
     DeclareProcedureName("finite?");
     checkArgumentLength(1);
-    argumentCheckReal(0, real);
-    if (real.isFlonum()) {
-        return Object::makeBool(real.toFlonum()->isFinite());
+    argumentCheckNumber(0, n);
+    if (n.isFlonum()) {
+        return Object::makeBool(n.toFlonum()->isFinite());
+    } else if (n.isCompnum()) {
+        Object r = n.toCompnum()->real();
+        if (r.isFlonum() && r.toFlonum()->isFinite()) {
+            Object i = n.toCompnum()->imag();
+            if (i.isFlonum() && i.toFlonum()->isFinite()) {
+                return Object::True;
+            }
+        }
+        return Object::False;
```

Comment:
This logic returns `#f` for `1+1i` (= `(make-rectangular #e1 #e1)` ).

Reply:
I may misunderstand but R7RS says
> he finite? procedure returns #t on all real numbers ex-
cept +inf.0, -inf.0, and +nan.0, and on complex num-
bers if their real and imaginary parts are both finite. Oth-
erwise it returns #f

I was guessin 1+1i returns #t.

Reply:
Wait.

Reply:
Thinking

Reply:
Thanks for the catch. I switched to implementation in Scheme. Hope it makes sense.